Python/initconfig.c

```diff
@@ -2944,6 +2945,11 @@ config_parse_cmdline(PyConfig *config, PyWideStringList *warnoptions,
             return _PyStatus_EXIT(0);
 
         case 'b':
+            if (!config->bytes_warning) {
+                fprintf(stderr,
+                        "-b option is deprecated since Python 3.15 "
+                        "and will be removed in Python 3.17\n");
```

Comment:
I would prefer to move this code at the end of pyinit_core() (before `done:` label). Example:

```c
    if (_Py_GetConfig()->bytes_warning) {
        fprintf(stderr,
                "-b option is deprecated since Python 3.15 "
                "and will be removed in Python 3.17\n");
    }
```

So it deprecates indirectly the `PyConfig.bytes_warning` member.